service/matching/taskQueueManager.go

```diff
@@ -711,6 +752,15 @@ func (c *taskQueueManagerImpl) newIOContext() (context.Context, context.CancelFu
 }
 
 func (c *taskQueueManagerImpl) userDataFetchSource() (string, error) {
+	if c.kind == enumspb.TASK_QUEUE_KIND_STICKY {
+		// Sticky queues get data from their corresponding normal queue
+		if c.normalName == "" {
+			// Older SDKs don't send the normal name. That's okay, they just can't use versioning.
+			return "", errMissingNormalQueueName
```

Comment:
We should make sure to not allow polls with no normal name and use versioning in the capabilities and fail with invalid argument in the frontend.